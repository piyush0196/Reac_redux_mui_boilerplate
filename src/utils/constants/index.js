import { EnglishIcon } from 'components/ui-kit/icons/iconComponents/EnglishIcon'
import { FrenchIcon } from 'components/ui-kit/icons/iconComponents/FrenchIcon'
import { GermanIcon } from 'components/ui-kit/icons/iconComponents/GermanIcon'
import { PortugalIcon } from 'components/ui-kit/icons/iconComponents/PortugalIcon'
import { RussianIcon } from 'components/ui-kit/icons/iconComponents/RussianIcon'
import { SpanishIcon } from 'components/ui-kit/icons/iconComponents/SpanishIcon'

/* ==========================================================================
  LANGUAGES
  ========================================================================== */
const LANGUAGES = [
  { label: 'English', languageCode: 'en', flag: EnglishIcon },
  { label: 'German', languageCode: 'de', flag: GermanIcon },
  { label: 'Spanish', languageCode: 'es', flag: SpanishIcon },
  { label: 'French', languageCode: 'fr', flag: FrenchIcon },
  { label: 'Portuguese', languageCode: 'pt', flag: PortugalIcon },
  { label: 'Russian', languageCode: 'ru', flag: RussianIcon }
]

/* ==========================================================================
  HTTP Method Types
========================================================================== */
const METHOD_TYPES = {
  get: 'get',
  post: 'post',
  put: 'put',
  delete: 'delete',
  patch: 'patch'
}

/* ==========================================================================
LocalStorage / Cookie data
========================================================================== */
const TOKEN = 'authtoken'
const LANGUAGE = 'language'

/* ==========================================================================
Loader types
========================================================================== */
const LOADER_TYPE = {
  SCALE: 'scale',
  PULSE: 'pulse'
}

/* ==========================================================================
  TOASTER / NOTIFICATION
========================================================================== */
const TOASTER_TYPE = {
  success: 'success',
  error: 'error',
  warning: 'warning',
  info: 'info'
}

/* ==========================================================================
  Loader Type Array
========================================================================== */
const LOADER_HANDLER_TYPES = {
  page: 'pageLoader',
  submit: 'submitButtonLoader',
  table: 'tableLoader'
}

/* ==========================================================================
  All the navigation route Paths
========================================================================== */
const gamesSectionParentRoute = '/games'

const ROUTE_PATHS = {
  // ----single routes------------
  rootPath: '/',
  login: '/login',
  settings: '/settings',
  dashboard: '/dashboard',
  profile: '/profile',
  changePassword: '/change-password',
  forgotPassword: '/forgot-password',
  // ----games section parent child routes------------
  games: gamesSectionParentRoute,
  perRoundReportReports: `${gamesSectionParentRoute}/pre-round-reports`,
  cancelledGames: `${gamesSectionParentRoute}/cancelled-games`
}

export {
  TOASTER_TYPE,
  LANGUAGES,
  TOKEN,
  LANGUAGE,
  ROUTE_PATHS,
  LOADER_TYPE,
  METHOD_TYPES,
  LOADER_HANDLER_TYPES
}
