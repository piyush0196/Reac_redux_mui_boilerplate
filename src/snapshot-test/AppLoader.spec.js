import { colors } from 'theme/colors'
import TestRenderer from 'react-test-renderer'
import { LOADER_TYPE } from 'utils/constants'
import AppLoader from 'components/ui-kit/AppLoader'

describe('Component - AppLoader', () => {
  const defaultProps = {
    size: 15,
    height: 40,
    width: 5,
    radius: 2,
    color: colors.darkNavyBlue,
    speedMultiplier: 1,
    margin: 2
  }

  it('renders PULSE variant AppLoader component correctly', () => {
    const wrapper1 = TestRenderer.create(
      <AppLoader variant={LOADER_TYPE.PULSE} {...defaultProps} />
    ).toJSON()
    expect(wrapper1).toMatchSnapshot()
  })

  it('renders SCALE variant AppLoader component correctly', () => {
    const wrapper2 = TestRenderer.create(
      <AppLoader variant={LOADER_TYPE.SCALE} {...defaultProps} />
    ).toJSON()
    expect(wrapper2).toMatchSnapshot()
  })
})
