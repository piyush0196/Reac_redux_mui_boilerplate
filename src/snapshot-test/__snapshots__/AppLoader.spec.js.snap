// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - AppLoader renders PULSE variant AppLoader component correctly 1`] = `
<span
  className="css-1xdhyk6"
>
  <span
    className="css-1xg279y"
  />
  <span
    className="css-b513mu"
  />
  <span
    className="css-xr14zd"
  />
</span>
`;

exports[`Component - AppLoader renders SCALE variant AppLoader component correctly 1`] = `
<span
  className="css-1xdhyk6"
>
  <span
    className="css-1m487lk"
  />
  <span
    className="css-1fqj8ti"
  />
  <span
    className="css-vs8qd2"
  />
  <span
    className="css-1adf12b"
  />
  <span
    className="css-110uc55"
  />
</span>
`;
